SCRIPT/openfpga_shell_script/skywater_generate_testbench_using_random_key_example_script.openfpga

```diff
@@ -0,0 +1,74 @@
+# This script is designed to generate Verilog testbenches
+# with a fixed device layout
+# It will only output netlists to be used by verification tools
+# including
+#   - Verilog testbenches, used by ModelSim 
+#   - SDC for a mapped FPGA fabric, used by Synopsys PrimeTime
+#
+#--write_rr_graph example_rr_graph.xml
+vpr ${VPR_ARCH_FILE} ${VPR_TESTBENCH_BLIF} --clock_modeling ideal --device ${OPENFPGA_VPR_DEVICE_LAYOUT} --route_chan_width ${OPENFPGA_VPR_ROUTE_CHAN_WIDTH} --absorb_buffer_luts off
+
+# Read OpenFPGA architecture definition
+read_openfpga_arch -f ${OPENFPGA_ARCH_FILE}
+
+# Read OpenFPGA simulation settings
+read_openfpga_simulation_setting -f ${OPENFPGA_SIM_SETTING_FILE}
+
+# Annotate the OpenFPGA architecture to VPR data base
+# to debug use --verbose options
+link_openfpga_arch --activity_file ${ACTIVITY_FILE} --sort_gsb_chan_node_in_edges
+
+# Check and correct any naming conflicts in the BLIF netlist
+check_netlist_naming_conflict --fix --report ./netlist_renaming.xml
+
+# Apply fix-up to clustering nets based on routing results
+pb_pin_fixup --verbose
+
+# Apply fix-up to Look-Up Table truth tables based on packing results
+lut_truth_table_fixup
+
+# Build the module graph
+#  - Enabled compression on routing architecture modules
+#  - Enable pin duplication on grid modules
+build_fabric --compress_routing --duplicate_grid_pin --generate_random_fabric_key #--verbose
```

Comment:
I suggest not to generate an random fabric key. 
- By default, OpenFPGA has a default built-in key. It can always work without any external key support. 
- In your case, a random key will be generated each time the OpenFPGA builds the fabric. It is never consistent. 
- I guess what you need to output the key to a file `--write_fabric_key <xml_file>`. And then you can use the key for all the testbench generation.

Reply:
switched to external key for 32x32